### DISCUSSION TITLE
TextExtensions.Text is showing warnings

### DISCUSSION THREAD

**Discussion by Ukt01** [QuestPDF/QuestPDF#728] [Q&A] [answered]
TextExtensions.Text(IContainer, object?)' is obsolete: 'This method has been deprecated since version 2022.12. Please use an overload where the text parameter is passed explicitly as a string.

 header.Cell().Element(CellStyle).AlignLeft().Text("Fee").SemiBold();
 header.Cell().Element(CellStyle).AlignRight().Text(Model.Fee);

**Comment by girlpunk** [comment]
What type is `Model.Fee`?

**Comment by MarcinZiabek** [comment, marked as answer]
It is completely expected. The newer library version does not perform any automated conversion from objects/numbers to text as usually, the default C# number formatting is different than ideal.

You can fix this in the following ways:

```csharp
// call the ToString method yourself
.Text(Model.Fee.ToString());

// or use formatting in string interpolation
.Text($"{Model.Fee:N2}");
```

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
